bach/tests/functional/bach/test_df_setitem.py

```
@@ -42,40 +51,64 @@ def check_set_const(engine, constants: List[Any], expected_series: Type[Series],
         ]
     )
 
+    for column_name in column_names:
+        assert isinstance(bt[column_name], expected_series)
+
+    if is_postgres(engine):
+        pg_types = {column_name: expected_pg_db_type for column_name in column_names}
+        assert_postgres_types(bt, pg_types)
```

I created a new function here rather than using the existing `assert_postgres_type`. This way we can do it in a similar way for Postgres and Athena, and check all types with one query. In the future we could then probably get rid of `assert_postgres_type` at some point.

Additional change is that we now check the type of the actual Series instance here, rather than the type associated with the database type. That seems more relevant for this test.

Now this function always does two queries, previously it would do `1 + number of columns added`.

---

A thought:
you now double the `if is_postgres` in both this function as well as in assert_postgres_type.
Would it be nice to silently return from the assert if the wrong engine was given, but check the assertion whet the right one is given? That way, this function becomes simpler and there is less duplicate checks.

---

I see your point. Reason I did it like this, is that last time I did something with `assert_postgres_type` I got the exact opposite feedback: https://github.com/objectiv/objectiv-analytics/pull/608#discussion_r843722206

Can't make everybody happy :confused:

---

ha, makes sense. Let's keep it like this then

---

https://github.com/objectiv/objectiv-analytics/pull/1130/files#r942236278 might help removing those doubled ifs ;)